### DISCUSSION TITLE
Start preview server with an https URL

### DISCUSSION THREAD

**Discussion by mkchandler** [statiqdev/Discussions#27] [Q&A]
Is there a way to start the preview server with an https URL when running `dotnet run -- preview`? I've read through the [`preview` docs](https://statiq.dev/web/running/preview-server) and see a setting to change the port, but nothing about http vs https.

This isn't stopping me from being able to run the server locally, I just ran into an issue where localhost ended up in the HSTS list in Chrome so I had to clear that out to be able to access the default "http://localhost:5080" URL.

**Comment by daveaglick** [comment]
After adding our corp domain at the day job to the HSTS preload list a while back, I'm well versed on the pain that HSTS can inflict :) But how the heck did you end up with localhost set as HSTS 😬? That sounds especially nightmare inducing.

Unfortunately there's no support for serving certificates in the Statiq preview server at the moment. What I've seen other folks do when they really need this is serve up the output folder from something like IIS or run [dotnet-serve](https://github.com/natemcmaster/dotnet-serve) after generation. I've had a long-standing plan to copy how dotnet-serve handles HTTPS but just haven't gotten around to it yet.

**Reply by mkchandler** [reply]
Sorry about the long wait on the reply, I somehow missed your reply to this over a month ago.

> After adding our corp domain at the day job to the HSTS preload list a while back, I'm well versed on the pain that HSTS can inflict :) But how the heck did you end up with localhost set as HSTS 😬? That sounds especially nightmare inducing.

I'm pretty sure it happened when running one of the default ASP.NET Core templates that ships with .NET Core. They have `app.UseHsts()` in their `Startup.cs` by default, and if you happen to run the app locally with an environment set to something other than "Development", that code will run. That's my best guess at least.

> Unfortunately there's no support for serving certificates in the Statiq preview server at the moment. What I've seen other folks do when they really need this is serve up the output folder from something like IIS or run [dotnet-serve](https://github.com/natemcmaster/dotnet-serve) after generation. I've had a long-standing plan to copy how dotnet-serve handles HTTPS but just haven't gotten around to it yet.

I'll take a look at one of these options. Thanks for the information!

**Reply by daveaglick** [reply]
No problem! Let me know how you get along - happy to continue working this with you if neither of those options works.
